_includes/pbc/all/install-features/202307.0/install-the-dynamic-store-prices-feature.md

```diff
@@ -0,0 +1,517 @@
+
+| TRANSFER | TYPE | EVENT | PATH |
+| --- | --- | --- | --- |
+| CurrencyCriteria | class | created | src/Generated/Shared/Transfer/CurrencyCriteriaTransfer  |
+| Store.defaultCurrencyIsoCode     | property | added | src/Generated/Shared/Transfer/CustomerTransfer  |
+
+{% endinfo_block %}
+
+### 3) Configure export to Redis
+
+1.  Set up publisher plugins:
+
+| PLUGIN | SPECIFICATION | PRERQUISITES | NAMESPACE |
+| --- | --- | --- | --- |
+| CurrencyStoreWritePublisherPlugin | Publishes currency store data to storage table. | None | Spryker\Zed\StoreStorage\Communication\Plugin\Publisher\CurrencyStore |
+
+
+**src/Pyz/Zed/Publisher/PublisherDependencyProvider.php**
+
+```php
+<?php
+
+namespace Pyz\Zed\Publisher;
+
+use Spryker\Zed\Publisher\PublisherDependencyProvider as SprykerPublisherDependencyProvider;
+use Spryker\Zed\StoreStorage\Communication\Plugin\Publisher\CurrencyStore\CurrencyStoreWritePublisherPlugin;
+
+class PublisherDependencyProvider extends SprykerPublisherDependencyProvider
+{
+    protected function getPublisherPlugins(): array
+    {
+        return array_merge(
+            // ...
+            $this->getStoreStoragePlugins(),
+        );
+    }
+    
+    /**
+     * @return array<\Spryker\Zed\PublisherExtension\Dependency\Plugin\PublisherPluginInterface>
+     */
+    protected function getStoreStoragePlugins(): array
+    {
+        return [
+            ...
+            new CurrencyStoreWritePublisherPlugin(),
+            ...
+        ];
+    }
+}
+```
+
+
+{% info_block warningBox "Verification" %}
+
+Ensure that, when a story currency added, updated, or deleted, it is exported to or removed from Storage.
+
+Storage type: Redis
+Target entity: Store
+
+Example expected data identifier: `kv:store:de`
+
+Example expected data fragment:
+
+```json
+{
+  "id_store": 1,
+  "name": "DE",
+  ...
+  "default_currency_iso_code": "EUR",
+  "available_currency_iso_codes": [
+    "CHF",
+    "EUR"
+  ],
+  ...
+}
+```
+
+{% endinfo_block %}
+
+
+
+### 4) Import data
+
+Import locale, store and country data:
+
+1.  Prepare your data according to your requirements using our demo data:
+
+Example for DE store currency-store configurations:
+**data/import/common/DE/currency_store.csv**
+
+```csv
+currency_code,store_name,is_default
+EUR,DE,1
+CHF,DE,0
+```
+
+| Column | REQUIRED | Data Type | Data Example | Data Explanation |
+| --- | --- | --- | --- | --- |
+| currency_code | mandatory | string | EUR | Define currency code. |
+|store_name |mandatory |string | DE | Define store name. |
+|is_default |mandatory |bool | 1 | Defines if the currency is default. |
+
+__Note: Only one currency can be default per store.__
+
+{% info_block warningBox “Verification” %}
+
+Make sure that:
+
+1.  The .csv files have an empty line in the end.
+2.  For each `currency_code` entry in csv files, there is a respective `code` entry in the table `spy_currency` in the database.
```

Comment:
do the `code` entries appear after importing or is this a prerequisite for importing?

Reply:
spy_currency is an old table and we already have an importer to fill it up. So we can say that his is a prerequisite